api/src/main/java/org/openmrs/module/attachments/AttachmentsServiceImpl.java

```diff
@@ -1,30 +1,48 @@
 package org.openmrs.module.attachments;
 
+import org.openmrs.Concept;
 import org.openmrs.Encounter;
 import org.openmrs.Obs;
 import org.openmrs.Patient;
 import org.openmrs.Person;
 import org.openmrs.Visit;
 
+import org.openmrs.api.ConceptService;
 import org.openmrs.api.context.Context;
 
 import org.openmrs.module.attachments.obs.Attachment;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
 
 import java.util.ArrayList;
 import java.util.List;
 
 public class AttachmentsServiceImpl implements AttachmentsService {
 	
+	@Autowired
+	@Qualifier(AttachmentsConstants.COMPONENT_ATT_CONTEXT)
+	protected AttachmentsContext context;
+	
+	@Autowired
+	@Qualifier("conceptService")
+	private ConceptService cs;
+	
 	@Override
 	public List<Attachment> getAttachments(Patient patient, Visit visit, Encounter encounter, boolean includeRetired) {
 		
 		List<Person> personList = new ArrayList<>();
 		List<Attachment> attachmentsList = new ArrayList<>();
 		personList.add(patient);
 		
+		List<String> conceptComplexList = context.getConceptComplexList();
+		List<Concept> questionConcepts = new ArrayList<>();
+		for (String uuid : conceptComplexList) {
+			questionConcepts.add(cs.getConceptByUuid(uuid));
```

Comment:
Please log an error when `cs.getConceptByUuid(uuid))` is null, which would mean that the concept is not found.

Reply:
@mks-d I added an error logging and submitted the changes. Thanks